### DISCUSSION TITLE
EXTENSIONS_AUTO_RELOAD does not reload extensions from workspaces

### DISCUSSION THREAD

**Discussion by YannickArmspach** [directus/directus#14922] [Archive]
Hello,

First of all, thanks for the great job! Directus is amazing :)

I'm looking to create the perfect environment for developing extensions.

I setup a monorepo with pnpm workspace and then NX. Everything is perfect for both except the extension auto reload. It is only triggered in the local extension directory, but not in dependencies from workspaces.

It would be interesting to define some paths for the "watch" like this:

```env
EXTENSIONS_PATH="./extensions"
EXTENSIONS_AUTO_RELOAD=true
EXTENSIONS_AUTO_RELOAD_PATH=["./extensions","../../packages/**/build"]
```

Currently I trying to add this functionality in ```api/src/extensions.ts```. Right way?

What do you think about this ? Do you have any recommendations for developpements?

Thanks,

### To Reproduce

- Create a monorepo with 2 workspaces: **apps** and **packages**
- install directus in **apps**
- create a directus extension in **packages**
- add your extention in the directus **package.json**
- run directus
- start extension in dev mode

**Comment by YannickArmspach** [comment]
#14939

**Comment by nickrum** [comment]
You can simply name your extension according to the [naming convention for package extensions](https://docs.directus.io/extensions/creating-extensions.html#publishing-your-extension) and install it into the Directus project package:

```json
{
  "dependencies": {
    "directus": "^9.17.1",
    "directus-extension-test": "workspace:*",
    "sqlite3": "^5.0.11"
  }
}
```

On startup, the Directus server will automatically load extensions installed this way. There was a bug which prevented auto-reloading from working but this was fixed in `directus@9.17.0`.

**Reply by arpadgabor** [reply]
> There was a bug which prevented auto-reloading from working but this was fixed in directus@9.17.0.

Can you detail this? Can't find anything in the changelog to explain. What did that fix do exactly?

**Reply by nickrum** [reply]
Sure! The relevant PR is #15209. Problem was that when starting Directus from the CLI, `ExtensionManager.initialize()` is called with `watch: false`. When it is called a second time in `createApp()`, the watcher is enabled, but the paths to the package extension entrypoints were never added to the watcher.
